lib/segment/src/types.rs

```diff
@@ -464,6 +486,22 @@ pub enum QuantizationConfig {
     Product(ProductQuantization),
 }
 
+impl QuantizationConfig {
+    /// Detect configuration mismatch against `other` that requires rebuilding
+    ///
+    /// Returns true only if both conditions are met:
+    /// - this configuration does not match `other`
+    /// - to effectively change the configuration, a quantization rebuild is required
+    pub fn mismatch_requires_rebuild(&self, other: &Self) -> bool {
+        match (self, other) {
+            (Self::Scalar(a), Self::Scalar(b)) => a.scalar.mismatch_requires_rebuild(&b.scalar),
+            (Self::Scalar(_), _) => true,
+            (Self::Product(a), Self::Product(b)) => a.product.mismatch_requires_rebuild(&b.product),
+            (Self::Product(_), _) => true,
+        }
+    }
```

Comment:
Ok, the documentation _does_ make sense for this method, but it's very confusing for the other two above.

Though, I think the first point should say something more specific, like "quantization __type__ has been changed" instead of "this configuration does not match `other`".